.github/workflows/devskim.yml

```
@@ -0,0 +1,34 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: DevSkim
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

if we want to run this tool on PRs, then we should not limit the branch here

---

Consider to use [Configuring default setup for code scanning](https://docs.github.com/en/code-security/code-scanning/enabling-code-scanning/configuring-default-setup-for-code-scanning-at-scale#configuring-default-setup-for-all-eligible-repositories-in-an-organization) instead of this PR.

---

Sadly, I don't have write permission so I cannot do this:<

---

I think it is currently triggered on push or pull request events on the "main" branch.
On which events exactly do you want it to fire?
```suggestion
    on:
        push:
        pull_request:
            branches: [ "main" ]
```
In this configuration, the push event doesn't specify any branches, which means that the workflow will be triggered on a push to any branch in the repository.

---

Would you like a weekly check?

---

```yaml
on:
  pull_request:
  push:
    branches:
      - main
```

Maybe we can do something like this? Run it for every PR and pushes to the main branch? Is the `schedule` key compulsory? I am wondering if we can remove it and do the check at the pace of PR and pushes.